x-pack/platform/plugins/shared/stack_alerts/server/rule_types/esql/fetch_esql_query.ts

```diff
@@ -201,3 +203,15 @@ function getGroupingObjects(results: Record<string, EsqlHit[]>, alertIdFields: s
   }
   return groupingObjectsPerResult;
 }
+
+function getFlappingFields(results: Record<string, EsqlHit[]>) {
```

Comment:
I followed a pattern here, but all three of these functions `sourceFieldsPerResult`, `groupingFieldsPerResult`, and `flappingFieldsPerResult` all iterate all over the same array so we probably need to combine them.